internal/loadbalancers/infrastructure/token_exchanger.go

```diff
@@ -0,0 +1,67 @@
+package infrastructure
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"time"
+
+	"golang.org/x/oauth2"
+)
+
+type TokenExchanger struct {
+	metalAPIKey string
+	client      *http.Client
+}
+
+var _ oauth2.TokenSource = (*TokenExchanger)(nil)
+
+func NewTokenExchanger(metalAPIKey string, client *http.Client) *TokenExchanger {
+	if client == nil {
+		client = http.DefaultClient
+	}
+	return &TokenExchanger{
+		metalAPIKey: metalAPIKey,
+		client:      client,
+	}
+}
+
+func (m TokenExchanger) Token() (*oauth2.Token, error) {
+	tokenExchangeURL := "https://iam.metalctrl.io/api-keys/exchange"
```

Comment:
Don't know that it would add much here, but if we had a spec for iam.metalcrtl.io we could use a generated client.

Reply:
cc @tylerauerbeck